tests/e2e/creation_test.go

```diff
@@ -286,11 +286,12 @@ func (tc *testContext) testAllApplicationCreation(t *testing.T) error { //nolint
 			err = tc.testApplicationCreation(c)
 
 			msg := fmt.Sprintf("error validating application %v when ", name)
-			if c.GetManagementState() == operatorv1.Managed {
-				require.NoError(t, err, msg+"enabled")
-			} else {
-				require.Error(t, err, msg+"disabled")
+
+			state := "enabled"
+			if c.GetManagementState() != operatorv1.Managed {
+				state = "disabled"
 			}
+			require.NoError(t, err, msg+state)
```

Comment:
maybe all-in-all we only need `require.NoError(t, err, msg+c.GetManagementState())` then it covers all cases even unmanaged in the future

Reply:
It changes a bit behaviour, but fine for me if it's ok.